bundles/org.openhab.binding.knx/src/test/java/org/openhab/binding/knx/internal/dpt/DPTTest.java

```diff
@@ -402,13 +404,81 @@ public void unitsValid(String unit) {
+        // but is still deterministic to get reproducible test results.
+        return IntStream.range(0, 3 * 256)
+                .mapToObj(i -> new byte[] { (byte) (i & 0xff), (byte) ((i / 2) & 0xff), (byte) ((i / 3) & 0xff) });
     }
 
     @ParameterizedTest
     @MethodSource("rgbValueProvider")
-    public void rgbTest(String value) {
-        Assertions.assertNotNull(ValueDecoder.decode("232.600", value.getBytes(StandardCharsets.UTF_8), HSBType.class));
+    public void dpt232BackToBackTest(byte[] value) {
+        backToBackTest232(value, "232.600");
+        backToBackTest232(value, "232.60000");
+    }
+
+    private void backToBackTest232(byte[] value, String dpt) {
+        // decode will apply transformation from raw bytes to String internally
+        HSBType hsb = (HSBType) ValueDecoder.decode(dpt, value, HSBType.class);
+        Assertions.assertNotNull(hsb);
+
+        // encoding will return a String in notation defined by Calimero: "r:xxx g:xxx b:xxx"
+        String result = ValueEncoder.encode(hsb, dpt);
+
+        // for back to back test, compare String representation
+        Assertions.assertEquals("r:" + (value[0] & 0xff) + " g:" + (value[1] & 0xff) + " b:" + (value[2] & 0xff),
+                result);
+    }
+
+    private static Stream<byte[]> xyYValueProvider() {
+        // Returning all combinations is too much. Implementation tries to catch rounding errors
+        // but is still deterministic to get reproducible test results.
+        //
+        // Cannot run make use of full numeric range, as colors cannot be represented in CIE space and
+        // back conversion would return different results.
+        // Use x: 0.1 .. 0.3, y: 0.3 .. 0.5 to be inside the triangle which can be converted without loss
+        return IntStream.range(77, 115).mapToObj(i -> new byte[] { (byte) ((i / 2) & 0xff), (byte) ((i) & 0xff),
+                (byte) (i & 0xff), (byte) ((i / 3) & 0xff), (byte) (((i - 50) * 3) & 0xff), 3 });
+        // last byte has value 3: c+b valid
+    }
+
+    @ParameterizedTest
+    @MethodSource("xyYValueProvider")
+    public void dpt242BackToBackTest(byte[] value) {
+        final String dpt = "242.600";
+
+        // decode will apply transformation from raw bytes to String internally
+        HSBType hsb = (HSBType) ValueDecoder.decode(dpt, value, HSBType.class);
+        Assertions.assertNotNull(hsb);
+
+        // encoding will return a String in notation defined by Calimero: "(x,xxxx y,yyyy) YY,Y %"
+        String result = ValueEncoder.encode(hsb, dpt);
+
+        // for back to back test, compare numerical values to allow tolerances
+        double dx = (((value[0] & 0xff) << 8) | (value[1] & 0xff)) / 65535.0;
+        double dy = (((value[2] & 0xff) << 8) | (value[3] & 0xff)) / 65535.0;
+        double dY = ((double) (value[4] & 0xff)) * 100.0 / 255.0;
+
+        final Pattern XYY_PATTERN = Pattern.compile(
```

Comment:
Is this the exact same pattern as above in `ValueDecoder`? If so, just remove the `private` from there and re-use it. This ensures changes are not missed in the future.

Reply:
@J-N-K done as requested